src/Handler/CurlFactory.php

```diff
@@ -460,14 +460,7 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf)
                 );
             }
             $conf[CURLOPT_NOPROGRESS] = false;
-            $conf[CURLOPT_PROGRESSFUNCTION] = function () use ($progress) {
-                $args = func_get_args();
-                // PHP 5.5 pushed the handle onto the start of the args
-                if (is_resource($args[0])) {
-                    array_shift($args);
-                }
-                call_user_func_array($progress, $args);
-            };
```

Comment:
If I am correct what guzzle library was doing here is to wrap one callable function inside another.

So this `call_user_func_array($progress, $args);` calls the developers callback function that he added from his options.

According to the docs the handler was not used and there fore that's why this part of the code exists:
```
                if (is_resource($args[0])) {
                    array_shift($args);
                }
```

If you ask why the handler does not exists. Most probably it's because guzzle provides an abstraction layer. Developers should not care about the handler that the guzzle internally uses.. So my guess is that they guzzle maintainers didn't want to have developers when setting this option having to think about the handler that the client internally uses.

Reply:
Yes, but it's for compatibility purpose as it doesn't exists in old php versions. It's also documented there.

The handle is eg. accessible here https://github.com/guzzle/guzzle/blob/003757426cdd50fbad1c34cdbcc8a55f70a4fe8b/src/Handler/EasyHandle.php#L17 public and not internal

Reply:
Can you leave this as unresolved? I believe that guzzle maintainers should share their toughs on this.